activitysim/abm/models/trip_scheduling.py

```diff
@@ -145,11 +170,18 @@ def update_tour_earliest(trips, outbound_choices):
     # set the trips "earliest" column equal to the max outbound departure
     # time for all inbound trips. preserve values that were used for outbound trips
     # FIXME - extra logic added because max_outbound_departure can be NA if previous failed trip was removed
-    tmp_trips["earliest"] = np.where(
-        ~tmp_trips["outbound"] & ~tmp_trips["max_outbound_departure"].isna(),
-        tmp_trips["max_outbound_departure"],
-        tmp_trips["earliest"],
-    )
+    if logic_version == 1:
+        tmp_trips["earliest"] = tmp_trips["earliest"].where(
+            tmp_trips["outbound"], tmp_trips["max_outbound_departure"]
+        )
+    elif logic_version > 1:
+        tmp_trips["earliest"] = np.where(
+            ~tmp_trips["outbound"] & ~tmp_trips["max_outbound_departure"].isna(),
+            tmp_trips["max_outbound_departure"],
+            tmp_trips["earliest"],
+        )
+    else:
+        raise ValueError(f"bad logic_version: {logic_version}")
```

Comment:
I think we need a check somewhere in here that will require logic_version == 2 if the scheduling mode is "relative", otherwise that mode will fail. Should mention this requirement in the documentation as well.

Reply:
I've added that check to the code, and a note in the documentation.

Reply:
Looks good, thanks @jpn-- .  (And thanks for fixing my spelling errors in the documentation too!)